casper/src/test/scala/coop/rchain/casper/api/ListeningNameAPITest.scala

```diff
@@ -1,193 +1,186 @@
-        block3 <- nodes(2).propagateBlock(deployDatas(2))(nodes: _*)
-        block4 <- nodes(0).propagateBlock(deployDatas(3))(nodes: _*)
-
-        listeningNameResponse2 <- blockApi.getListeningNameDataResponse(depth, listeningName)
-        _ = inside(listeningNameResponse2) {
-          case Right((blockResults, l)) =>
-            val data2   = blockResults.map(_.postBlockData)
-            val blocks2 = blockResults.map(_.block)
-            data2 should be(
-              List(
-                List(resultData, resultData, resultData, resultData),
-                List(resultData, resultData, resultData),
-                List(resultData, resultData),
-                List(resultData)
-              )
-            )
-            blocks2.length should be(4)
-            l should be(4)
-        }
-        block5 <- nodes(1).propagateBlock(deployDatas(4))(nodes: _*)
-        block6 <- nodes(2).propagateBlock(deployDatas(5))(nodes: _*)
-        block7 <- nodes(0).propagateBlock(deployDatas(6))(nodes: _*)
-
-        listeningNameResponse3 <- blockApi.getListeningNameDataResponse(depth, listeningName)
-        _ = inside(listeningNameResponse3) {
-          case Right((blockResults, l)) =>
-            val data3   = blockResults.map(_.postBlockData)
-            val blocks3 = blockResults.map(_.block)
-            data3 should be(
-              List(
-                List(
-                  resultData,
-                  resultData,
-                  resultData,
-                  resultData,
-                  resultData,
-                  resultData,
-                  resultData
-                ),
-                List(resultData, resultData, resultData, resultData, resultData, resultData),
-                List(resultData, resultData, resultData, resultData, resultData),
-                List(resultData, resultData, resultData, resultData),
-                List(resultData, resultData, resultData),
-                List(resultData, resultData),
-                List(resultData)
-              )
-            )
-            blocks3.length should be(7)
-            l should be(7)
-        }
-        listeningNameResponse3UntilDepth <- blockApi.getListeningNameDataResponse(1, listeningName)
-        _ = inside(listeningNameResponse3UntilDepth) {
-          case Right((_, l)) => l should be(1)
-        }
-        listeningNameResponse3UntilDepth2 <- blockApi.getListeningNameDataResponse(2, listeningName)
-        _ = inside(listeningNameResponse3UntilDepth2) {
-          case Right((_, l)) => l should be(2)
-        }
-      } yield ()
+  it should "return empty result if listening name deeper than expected" in {
+    implicit val (log, sp, rm, bs, bds) = createMocks[Task]
+    for {
+      blockApi <- createBlockApi[Task]("root", 50, createValidator.some)
+      res      <- blockApi.getListeningNameDataResponse(1, listeningName)
+    } yield {
+      res shouldBe 'right
+      val (_, length) = res.value
+      length shouldBe 0
```

Comment:
You can check value directly. `.value` will be reported in a nice way even when is result is Left.
```scala
res.value shouldBe (_, 0)
```

Reply:
[Fixed](https://github.com/rchain/rchain/pull/3787/commits/12138a8bb317c3cb8d1b37d1a426bae856a00c63) with small change to satisfy compiler.

Reply:
In that case you can just use `(Seq(), 0)` which is even more correct.